### PR TITLE
fix: Implement custom 6492 compatible signMessage

### PR DESCRIPTION
**Pull request by ch4r10t33r**
## Description

Added `signMessage` function that creates EIP-6492 compatible signatures for EIP-7702 wallets, allowing message signing before and after smart account delegation.

## Changes
- ✅ Added `signMessage()` method to `EtherspotTransactionKit` class
- ✅ Updated `IInitial` interface to include `signMessage` signature
- ✅ Added comprehensive unit tests (7 test cases)
- ✅ Created example program demonstrating usage

## How Has This Been Tested?

Unit testing
```bash
npm test -- --testNamePattern="signMessage"
```

## Example Output

```
============================================================
Example 1: Sign Message with Non-Delegated EOA
============================================================

EOA Address: 0x1234...
Is Delegated: No
Message to Sign: Hello, World! This is a test message.
EIP-6492 Format Valid: ✓ Yes
Magic Prefix: 0x6492
Signature: 0x6492abc123...
Total Signature Length: 400+ chars
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Summary by CodeRabbit

* **New Features**
  * Added signMessage() for EIP-6492-compatible signatures on EIP-7702 delegated EOA wallets
  * Automatically creates EIP-7702 authorization if EOA is not yet deployed
  * Supports text and hex messages; signatures compatible with EIP-6492 validators
  * Available only in delegated EOA wallet mode

* **Tests**
  * Expanded tests covering signMessage flows, edge cases, and error paths

* **Changelog**
  * Bumped release to 2.1.4 and added entry detailing the above features

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by RanaBug — DISMISSED** [review]

File: lib/TransactionKit.ts [hunk 1]
```diff
@@ -656,6 +663,202 @@ export class EtherspotTransactionKit implements IInitial {
...
+
+          authorization = await bundlerClient.signAuthorization({
+            account: owner,
+            contractAddress: delegateAddress,
+          });
+        }
+
+        if (!authorization) {
+          throw new Error('Authorization is required but was not created');
+        }
```
Comment (IAmKio) [inline lib/TransactionKit.ts:743]: In what situation would this happen?

**Review by IAmKio — DISMISSED** [review]
Just a minor comment but other than that it's not a showstopped - LGTM

File: lib/TransactionKit.ts (hunk 1 above)
Comment (RanaBug) [reply lib/TransactionKit.ts:743]: Then signMessage fails, but when a we use signMessage method, it needs to be added in a try/catch

Comment (RanaBug) [reply lib/TransactionKit.ts:743]: added extra info on this error

**Review by RanaBug — APPROVED** [review]
